### PR TITLE
Return status 1 on invalid/incomplete commands.

### LINKED ISSUE #2146: `singularity <unknown command>` returns 0 for an exit status
## Version of Singularity:

$ singularity --version
singularity version v3.0.0-alpha.2-310-g3812cec2-dirty

### Expected behavior

$ singularity bob
$ echo $?
1

### Actual behavior

$ singularity bob
$ echo $?
0

### Steps to reproduce behavior

See above.

This is causing a few false positives with the integration tests. At the very least the `selftest` command, which is not implemented, is passing when it's test runs.

### PR DESCRIPTION
**Pull request by phpHavok**
**Description of the Pull Request (PR):**

This PR causes singularity to exit with status code 1 (rather than 0) when given invalid commands. This has been tested only for the commands which accept sub-commands: (singularity, singularity instance, singularity capability, singularity keys). The SilenceErrors directive prevents the error from being printed to the screen. I'm not sure if there are any unintended consequences of this for e.g. debugging. A couple tests had to be modified to expect this new failure.

**This fixes or addresses the following GitHub issues:**

- Fixes #2146

Attn: @singularity-maintainers

### CODE REVIEW

File: cmd/singularity/cli/capability.go [hunk 1]
```diff
@@ -43,13 +44,16 @@ func init() {
...
 
-	Use:     docs.CapabilityUse,
-	Short:   docs.CapabilityShort,
-	Long:    docs.CapabilityLong,
-	Example: docs.CapabilityExample,
+	Use:           docs.CapabilityUse,
+	Short:         docs.CapabilityShort,
+	Long:          docs.CapabilityLong,
+	Example:       docs.CapabilityExample,
+	SilenceErrors: true,
```
Comment (jscook2345) [inline cmd/singularity/cli/capability.go:56]: What is the purpose of SilenceErrors here? Should this be instead SilenceUsage?

Comment (phpHavok) [reply cmd/singularity/cli/capability.go:56]: I wasn't sure what the desired outcome was. I assumed the expected usage was to return 1 but still show usage, so I silenced the errors. I can swap that around if you prefer the other way.

Also, is it correct (as I've implemented it) to return an error on a blank "singularity" command?

File: cmd/singularity/cli_test.go [hunk 2]
```diff
@@ -29,9 +29,9 @@ func TestSelfTest(t *testing.T) {
 	defer test.ResetPrivilege(t)
 
 	cmd := exec.Command(cmdPath, "selftest")
-	if b, err := cmd.CombinedOutput(); err != nil {
+	if b, err := cmd.CombinedOutput(); err == nil {
 		t.Log(string(b))
-		t.Fatalf("unexpected failure running selftest: %v", err)
+		t.Fatal("selftest passed, but it isn't implemented?")
```
Comment (ArangoGutierrez) [inline cmd/singularity/cli_test.go]: I still prefer the message 
```
unexpected failure running <something>
```
for consistency across the testing suite

File: cmd/singularity/help_test.go [hunk 3]
```diff
@@ -15,21 +15,26 @@ import (
...
 		t.Run(tt.name, test.WithoutPrivilege(func(t *testing.T) {
 			cmd := exec.Command(cmdPath, tt.argv...)
-			if b, err := cmd.CombinedOutput(); err != nil {
+			b, err := cmd.CombinedOutput()
+			if err != nil && tt.shouldPass {
 				t.Log(string(b))
 				t.Fatalf("unexpected failure running '%v': %v", strings.Join(tt.argv, " "), err)
+			} else if err == nil && !tt.shouldPass {
+				t.Log(string(b))
+				t.Fatalf("Test %s passed, but was expected to fail.", tt.name)
```
Comment (ArangoGutierrez) [inline cmd/singularity/help_test.go]: Same comment as above 
```
unexpected failure running <cmd>
```
or 
```
unexpected success running <cmd>
```
for consistency

**Review by ArangoGutierrez — CHANGES_REQUESTED** [review]
Looks good for me, just some err messages changes for consistency across the test suite.

File: cmd/singularity/help_test.go (hunk 3 above)
Comment (phpHavok) [reply cmd/singularity/help_test.go]: Fixed!

File: cmd/singularity/cli_test.go (hunk 2 above)
Comment (phpHavok) [reply cmd/singularity/cli_test.go]: Fixed!

**Review by ArangoGutierrez — APPROVED** [review]

**Review by bauerm97 — CHANGES_REQUESTED** [review]
Hey @phpHavok 

First off, thanks for this contribution it's greatly appreciated. Everything here looks good on my end with your patch. In testing the return codes of all `singularity` commands, I did however find that `singularity instance stop` returns 0 even when run with no arguments. I looked into the `instance_stop.go` file briefly and I think the culprit is because that command can work with 0 arguments if a specific flag is set. Would you be willing to patch that command as well? I think the fix is as simple as changing it to a `RunE:` func, and returning an error after `cmd.Usage()`. That way all return codes should be consistent across the board.

Cheers!

**Comment by phpHavok** [comment]
@bauerm97 Thanks for the comments, and no problem. I'm glad to be working on this project and getting to know you all.

I've made the change you requested and am running unit tests now. If they pass, I'll push so that CI can run here in a few minutes. The usage is actually automatically shown when RunE returns an error, so I removed the usage line as well. I think it works as intended now.

**Review by bauerm97 — APPROVED** [review]
LGTM, thanks for the contribution

**Pull request merged**